web/modules/custom/rsvplist/src/Controller/ReportController.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Drupal\rsvplist\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+
+/**
+ * ReportController class to process the routes.
+ */
+class ReportController extends ControllerBase {
+
+  /**
+   * Function to load the RSVP submission datas
+   * that includes the user name of the user submitting the form
+   * Node id of the node
+   * and the email id the form was submitted with.
+   *
+   * @return array|null
+   *   Returns the array.
+   */
+  protected function load() {
+
+    try {
+      $database = \Drupal::database();
+      $query = $database->select('rsvplist', 'r');
+      // Joining the User's table to get the information about the user's
+      // username.
+      $query->join('users_field_data', 'u', 'r.uid = u.uid');
+      // Joining with the node table to get the information about the Event's
+      // name.
+      $query->join('node_field_data', 'n', 'r.nid = n.nid');
+
+      // Add the Fields to be displayed.
+      $query->addField('u', 'name', 'username');
+      $query->addField('n', 'title');
+      $query->addField('r', 'mail');
+
+      // Executing the query and fetching the result.
+      $result = $query->execute()->fetchAll(\PDO::FETCH_ASSOC);
+
+      return $result;
+    }
+    catch (\Exception $e) {
+      \Drupal::messenger()->addMessage($this->t("Sorry Data Couldn't be loaded"));
+      return NULL;
+    }
+  }
+
+  /**
+   * Function to Return an render array of all the RSVP Reports
+   *
+   * @return array|null
```

Comment:
Only array is returned why array|null is used? Even if the table content is empty the message will be returned.

Reply:
@Vishalprasad28